### PR TITLE
Adapt index scraper to new MSCI layout

### LINKED ISSUE #18: update `get_index_regions` to adapt to new MSCI website layout
MSCI has changed their website layout, and the function `get_index_regions` now errors because of this.

### PR DESCRIPTION
**Pull request by jdhoffa**
Closes #18

``` r
library(pacta.data.scraping)

get_index_regions()
#> # A tibble: 95 × 3
#>    equity_market country     country_iso
#>    <chr>         <chr>       <chr>      
#>  1 GlobalMarket  Canada      CA         
#>  2 GlobalMarket  Austria     AT         
#>  3 GlobalMarket  Italy       IT         
#>  4 GlobalMarket  Australia   AU         
#>  5 GlobalMarket  US          US         
#>  6 GlobalMarket  Belgium     BE         
#>  7 GlobalMarket  Netherlands NL         
#>  8 GlobalMarket  Hong Kong   HK         
#>  9 GlobalMarket  Denmark     DK         
#> 10 GlobalMarket  Norway      NO         
#> # ℹ 85 more rows
```

<sup>Created on 2023-06-21 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### CODE REVIEW

File: R/get_index_regions.R [hunk 1]
```diff
@@ -57,3 +57,21 @@ get_index_regions <- function() {
   dplyr::distinct() %>%
   dplyr::arrange("equity_market", "country", "country_iso")
 }
+
+fetch_url_html <- function(url) {
+
+  # start the headless browser and capture the DOM as HTML after JavaScript runs
+  session <- chromote::ChromoteSession$new()
+  session$Network$setUserAgentOverride(userAgent = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.4 Safari/605.1.15")
```
Comment (AlexAxthelm) [inline R/get_index_regions.R:65]: Non-blocking: Is this the right useragent for us to use as a scraper?

**Review by AlexAxthelm — APPROVED** [review]
This gives consistent results with the prior version?

**Comment by jdhoffa** [comment]
> This gives consistent results with the prior version?

It's very hard for me to assess this, since we never actually write the `index_regions` file to disk. We just use it in the `dataprep_abcd_scen_connection` function...

File: R/get_index_regions.R (hunk 1 above)
Comment (jdhoffa) [reply R/get_index_regions.R:65]: Genuinely no idea, I just followed suit with what @cjyetman had initially done.

What factors influence the choice of useragent?

**Pull request merged**